SALTO-6322: field to context weak reference

IdoZakk opened the pull request:
Turn the references from field to contexts to be weak, based on the projects
---

If a field is deployed all its contexts will be dependencies. If one of those contexts is scoped specific, and none of the projects exist in the target env, the context will turn to global context and that is not the desired behavior 

---
_Release Notes_: 
None

---
_User Notifications_: 
None

shayc331 commented on packages/jira-adapter/test/weak_references/field_contexts.test.ts:
```diff
@@ -0,0 +1,173 @@
+    })
+  })
+  describe('findWeakReferences', () => {
+    it('should return weak references projects', async () => {
```
please fix the text, it is not about projects :)

shayc331 commented on packages/jira-adapter/test/weak_references/field_contexts.test.ts:
```diff
@@ -0,0 +1,173 @@
+        contextInstanceSomeValid,
+      ])
+    })
+    it('should remove the invalid context', async () => {
```
IIUC the contexts were removed here because their referenced project was not found in the elementsSource. can you please add tests for projects that were found in the elementsSource? a test that involves `removeMissingContextProjects`

IdoZakk replied:
It is done (the coverage is 100%). I will talk to you f2f

IdoZakk replied:
Following our f2f- it is covered in another PR

shayc331 reviewed (approved):
looking good, had a small comment about the tests
approving to save a cycle

IdoZakk commented:
Following a discussion with @kami-delaroz we removed the fix element warnings for contexts with no valid projects in the target env. Instead I added a change validator to block them from deployment

shayc331 reviewed (commented):
looking good! 
had one comment
